utf8.h

```diff
@@ -104,15 +104,15 @@ typedef char utf8_int8_t;
 
 /* Return less than 0, 0, greater than 0 if src1 < src2, src1 == src2, src1 >
  * src2 respectively, case insensitive. */
-utf8_constexpr14 utf8_nonnull utf8_pure int
+utf8_constexpr14 utf8_nonnull int
```

Comment:
Sorry colour me dumb here - why is this not pure?

Reply:
Because it's calling utf8codepoint which is a non-pure function (utf8codepoint is writing to the pointer it receives as a parameter). This results in undefined behavior which is not very well handled by clang.

Reply:
Oh that's annoying. I might just change that instead!

Reply:
You mean the definition of utf8codepoint? I think that would require something like:
```
utf8_constexpr14_impl struct { utf8_int8_t *str; utf8_int32_t out_codepoint; }
utf8codepoint(const utf8_int8_t *utf8_restrict str) 
```

so that the function becomes pure.